source/contracts/reporting/ReputationToken.sol

```diff
@@ -108,6 +116,36 @@ contract ReputationToken is DelegationTarget, Extractable, ITyped, Initializable
         return universe;
     }
 
+    function getTotalMigrated() public view returns (uint256) {
+        return totalMigrated;
+    }
+
+    function updateSiblingMigrationTotal(IReputationToken _token) public returns (bool) {
+        require(_token != this);
+        IUniverse _supposedUniverse = _token.getUniverse();
+        require(_token == universe.getParentUniverse().getChildUniverse(_supposedUniverse.getParentPayoutDistributionHash()).getReputationToken());
+        totalTheoreticalSupply += migratedToSibling[_token];
+        migratedToSibling[_token] = _token.getTotalMigrated();
+        totalTheoreticalSupply -= migratedToSibling[_token];
```

Comment:
totalTheoreticalSupply gets added to then subtracted from, not following reason why.

Reply:
These functions exist to allow someone off chain to request that the theoretical supply be adjusted because of new information. In this case they are indicating that REP was migrated to a sibling universe and asking that the theoretical supply be lowered accordingly.

Since we may have already accounted for this siblings migration total in the past we need to first add back whatever we previously subtracted, then update the amount we're using in the calculation, then subtract it.

Reply:
ok, so the total theoretical supply is essentially rolled back then updated based on the migrated rep to sib universe.

Reply:
Yup! We have to do this since we can't trust the off chain provider to give us a delta. We have to recompute basedo n what the old value was and check what it actually is now.

Reply:
I'm not sure if we care, but this could be done with fewer storage updates with something like:
```
uint256 newMigrations = _token.getTotalMigrated() - migratedToSibling[_token];
migratedToSibling[_token] += newMigrations;
totalTheoreticalSupply -= newMigrations;
```